### DISCUSSION TITLE
[Mobile] Caret is obliterated by colored text background

### DISCUSSION THREAD

**Discussion by jongkb** [singerdmx/flutter-quill#300] [Mobile]
(Observed on Android)

**Actual**
Caret appears behind colored text background, making the caret partially visible.

![Screenshot_20210517-124431](https://user-images.githubusercontent.com/1127628/118433699-2601a800-b70e-11eb-8555-a1f3defc30ee.png)

**Expected**
Caret appears on top of colored text background.

Thank you.

**Comment by singerdmx** [comment]
![image](https://user-images.githubusercontent.com/122956/125176610-91ee1300-e189-11eb-96fa-4d854c5d403c.png)
 code is here

**Comment by singerdmx** [comment]
This is basically how to add background color for TextSpan that will not be obliterated by cursor

@florianh01 @friebetill  any thought?

**Comment by xinyuezhang0402** [comment]
Likely the `canvas.drawRRect` method

![image](https://user-images.githubusercontent.com/40752995/125179191-e5b82680-e1a0-11eb-87f4-fc6b3c14a42f.png)

**Comment by singerdmx** [comment]
You are right
Need to figure out how canvas works

**Comment by singerdmx** [comment]
Probably has to do with `LayerLink`

![image](https://user-images.githubusercontent.com/122956/125181790-38074080-e1bd-11eb-8a95-73fdb179080f.png)

**Comment by huqigu** [comment]
<img width="376" alt="image" src="https://github.com/singerdmx/flutter-quill/assets/10161096/ff1ad1be-7d4d-4f0b-9ab2-e07b83a80ba2">
I set this property to true and it will work. On Android, this property defaults to false.
